test/FunctionalTests/Microsoft.OData.Core.Tests/JsonLight/TestODataJsonLightWriterResourceState.cs

```diff
@@ -0,0 +1,58 @@
+﻿//---------------------------------------------------------------------
+// <copyright file="TestODataJsonLightWriterResourceState.cs" company="Microsoft">
+//      Copyright (C) Microsoft Corporation. All rights reserved. See License.txt in the project root for license information.
+// </copyright>
+//---------------------------------------------------------------------
+
+using System;
+using Microsoft.OData.Edm;
+using Microsoft.OData.Evaluation;
+using Microsoft.OData.JsonLight;
+
+namespace Microsoft.OData.Tests.JsonLight
+{
+    internal class TestODataJsonLightWriterResourceState : IODataJsonLightWriterResourceState
+    {
+        public TestODataJsonLightWriterResourceState(
+            ODataResourceBase resource,
+            IEdmStructuredType structuredType,
+            ODataResourceSerializationInfo serializationInfo,
+            IEdmNavigationSource navigationSource,
+            bool isUndeclared)
+        {
+            Resource = resource;
+            ResourceType = structuredType;
+            ResourceTypeFromMetadata = structuredType;
+            SerializationInfo = serializationInfo;
+            NavigationSource = navigationSource;
+            IsUndeclared = isUndeclared;
+        }
+
+        public ODataResourceBase Resource { get; }

```

Comment:
Are we using `ODataResourceBase` since multiple classes inherits from `ODataResourceBase`?
e.g `ODataResource`, `ODataDeletedResource`

Reply:
Yes. Makes the class reusable by all derived classes